lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -238,145 +274,248 @@ bool Caffe2ModelLoader::hasMultidirectionalBroadcast(
   return false;
 }
 
-llvm::Error Caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
-  ArgumentDictionaryTy dict = loadArgumentMap(op);
+llvm::Error Caffe2ModelLoader::loadConv(const caffe2::OperatorDef &op,
+                                        ArgumentDictionaryTy &dict) {
+  const std::string &opName = loadOperatorName(op);
   const std::string &typeName = op.type();
 
-  // Check if operator is supported in parent class, CommonOperatorLoader.
-  bool loadCommonOperatorSuccess;
-  ASSIGN_VALUE_OR_RETURN_ERR(loadCommonOperatorSuccess,
-                             tryLoadCommonOperator(typeName, op, dict));
-  if (loadCommonOperatorSuccess) {
-    return llvm::Error::success();
+  // Load the inputs:
+  std::vector<unsigned_t> strides;
+  ASSIGN_VALUE_OR_RETURN_ERR(strides, getSizeHW(dict, "stride", 1));
+  std::vector<unsigned_t> pads;
+  ASSIGN_VALUE_OR_RETURN_ERR(pads, getPads(dict));
+  std::vector<unsigned_t> kernels;
+  ASSIGN_VALUE_OR_RETURN_ERR(kernels, getSizeHW(dict, "kernel", 0));
+  unsigned_t group = 1;
+  if (dict.count("group")) {
+    ASSIGN_VALUE_OR_RETURN_ERR(group, loadInt(dict["group"]));
+  }
+  std::string order = "NCHW";
+  if (dict.count("order")) {
+    ASSIGN_VALUE_OR_RETURN_ERR(order, loadStr(dict["order"]));
   }
-  const std::string &opName = loadOperatorName(op);
 
-  if (typeName == "Conv" || typeName == "ConvRelu" || typeName == "Int8Conv" ||
-      typeName == "Int8ConvRelu") {
-    // Load the inputs:
-    std::vector<unsigned_t> strides;
-    ASSIGN_VALUE_OR_RETURN_ERR(strides, getSizeHW(dict, "stride", 1));
-    std::vector<unsigned_t> pads;
-    ASSIGN_VALUE_OR_RETURN_ERR(pads, getPads(dict));
-    std::vector<unsigned_t> kernels;
-    ASSIGN_VALUE_OR_RETURN_ERR(kernels, getSizeHW(dict, "kernel", 0));
-    unsigned_t group = 1;
-    if (dict.count("group")) {
-      ASSIGN_VALUE_OR_RETURN_ERR(group, loadInt(dict["group"]));
-    }
-    std::string order = "NCHW";
-    if (dict.count("order")) {
-      ASSIGN_VALUE_OR_RETURN_ERR(order, loadStr(dict["order"]));
-    }
+  NodeValue in;
+  ASSIGN_VALUE_OR_RETURN_ERR(in, getNodeValueByName(op.input(0)));
+
+  Constant *w;
+  ASSIGN_VALUE_OR_RETURN_ERR(w, getConstantByName(op.input(1)));
+
+  // Transpose the weights to the right format. Glow expects to read the
+  // weights in the format CRSK.
+  // C - output_depth, R - filter_height, S - filter_width, K - input_depth.
+  // Caffe2 "Conv" op always stores the weight as CKRS.
+  Tensor wT;
+  w->getPayload().transpose(&wT, NCHW2NHWC);
+  w = G_.getParent()->createConstant(w->getName(), std::move(wT));
```

Comment:
Same as other comment, I'd rather not create this new Constant here, and instead defer to graph optimizations.